src/nvim/sign.c

```diff
@@ -441,6 +435,27 @@ static int sign_define_by_name(char *name, char *icon, char *text, char *linehl,
     }
   }
 
+  // Update already placed signs and redraw if necessary when modifying a sign.
+  if (!new_sign) {
+    bool should_redraw = true;
+    for (size_t i = 0; i < kv_size(decor_items); i++) {
+      if (strequal(name, kv_A(decor_items, i).sign_name)) {
```

Comment:
Check for `kSHIsSign` first?

Reply:
Also, `kv_A(decor_items, i)` is used a lot of times here, so it may be worth assigning it to a variable.

Reply:
> Check for `kSHIsSign` first?

`strequal` checks if `sign_name` is non-NULL first, which is a smaller set (i.e. legacy signs only) for which `kSHIsSign` should always be set.